hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/snapshot/OMSnapshotMoveDeletedKeysResponse.java

```diff
@@ -78,42 +86,74 @@ public OMSnapshotMoveDeletedKeysResponse(@Nonnull OMResponse omResponse) {
   protected void addToDBBatch(OMMetadataManager omMetadataManager,
       BatchOperation batchOperation) throws IOException {
 
-    if (nextSnapshot != null) {
-      RDBStore nextSnapshotStore =
-          (RDBStore) nextSnapshot.getMetadataManager().getStore();
-      // Init Batch Operation for snapshot db.
-      try (BatchOperation writeBatch = nextSnapshotStore.initBatchOperation()) {
-        processKeys(writeBatch, nextSnapshot.getMetadataManager());
-        processDirs(writeBatch, nextSnapshot.getMetadataManager());
-        nextSnapshotStore.commitBatchOperation(writeBatch);
-        nextSnapshotStore.getDb().flushWal(true);
-        nextSnapshotStore.getDb().flush();
+    // Note: a trick to get
+    // To do this properly, refactor OzoneManagerDoubleBuffer#addToBatch and
+    // add OmSnapshotManager as a parameter.
```

Comment:
Couldn't the snapshotManager just be added as a parameter to the OMSnapshotMoveDeletedKeysResponse constructor, similiar to how it is done for the secretManager here: https://github.com/apache/ozone/blob/016a433a06506714dcfea5ccb806f758770135e1/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/s3/security/OMSetSecretResponse.java#L51-L60

Reply:
If I hadn't changed `OmMetadataManagerImpl` constructor I might have taken this "lazy" approach. IMO that doesn't seem to be a memory-efficient way to pass along something that could be retrieved from the active OM. It is occupying 64 bits for each request that are not flushed in the double buffer.

Reply:
ok